app_dart/lib/src/service/branch_service.dart

```diff
@@ -63,6 +75,44 @@ class BranchService {
     }
   }
 
+  /// Creates a flutter/recipes branch that aligns to a flutter/flutter branch.
+  ///
+  /// Take the example repo history:
+  ///   flutter/flutter: A -> B -> C -> D -> E
+  ///   flutter/recipes: V -> W -> X -> Y -> Z
+  ///
+  /// If flutter/flutter branches at C, this finds the flutter/recipes commit that should be used for C.
+  /// The best guess for a flutter/recipes commit that aligns with C is whatever was the most recently committed
+  /// before C was committed.
+  ///
+  /// Once the flutter/recipes commit is found, it is branched to match flutter/flutter.
+  ///
+  /// Generally, this should work. However, some edge cases may require CPs. Such as when commits land in a
+  /// short timespan, and require the release manager to CP onto the recipes branch (in the case of reverts).
+  Future<void> branchFlutterRecipes(String branch) async {
+    final RepositorySlug recipesSlug = RepositorySlug('flutter', 'recipes');
+    if ((await gerritService.branches('${recipesSlug.owner}-review.googlesource.com', recipesSlug.name,
+            subString: branch))
+        .contains(branch)) {
+      // subString is a regex, and can return multiple matches
+      log.warning('$branch already exists for $recipesSlug');
+      throw BadRequestException('$branch already exists');
+    }
+    final Iterable<GerritCommit> recipeCommits = await gerritService.commits(recipesSlug, 'main');
+    log.info('$recipesSlug commits: $recipeCommits');
+    final GithubService githubService = await config.createDefaultGitHubService();
+    final List<RepositoryCommit> githubCommits = await githubService.listCommits(Config.flutterSlug, branch, null);
+    log.info('${Config.flutterSlug} branch commits: $githubCommits');
+    for (GerritCommit recipeCommit in recipeCommits) {
```

Comment:
A binary search would be a good fit to find the target, especially when the list is long.

Reply:
I tried implementing it using [package:collections binarySearch](https://pub.dev/documentation/collection/latest/collection/binarySearch.html) but it doesn't really work given we want it based on a DateTime before, not exactly equal.

From a complexity standpoint, since this will be automated, the average run is going to finish after 3 passes as flutter/recipes doesn't get many commits.

Reply:
ahhh that sounds like my expertise. I can write a customized bsearch helper function if we want a bsearch approach?

Reply:
If the list is short enough, this shouldn't be a big deal. You make the calls.

Reply:
Thanks @XilaiZhang but no need. Comparing the trade offs of code complexity vs speed, it's preferable to write maintainable code first.